backend/tournesol/views/unconnected_entities.py

```diff
@@ -0,0 +1,81 @@
+"""
+API endpoints to show unconnected entities
+"""
+from collections import defaultdict
+
+from django.db.models import ObjectDoesNotExist
+from django.http import Http404
+from drf_spectacular.utils import extend_schema, extend_schema_view
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+
+from tournesol.views.mixins.poll import PollScopedViewMixin
+
+from ..models import Comparison, Entity
+from ..serializers.entity import EntitySerializer
+
+
+@extend_schema_view(
+    get=extend_schema(description="Show unconnected entities for the current user")
+)
+class UnconnectedEntitiesView(
+    PollScopedViewMixin,
+    generics.ListAPIView
+):
+    """
+    API view for showing unconnected entities
+    """
+    serializer_class = EntitySerializer
+    permission_classes = [IsAuthenticated]
+
+    __already_visited_node__ = set()
+    __all_entites__ = set()
+    __all_connections__ = defaultdict(set)
+
+    def get_related_entities(self, entity_id):
+
+        self.__already_visited_node__.add(entity_id)
+        self.__all_entites__.remove(entity_id)
+
+        related_entities = set()
+        for other_node in self.__all_connections__.get(entity_id):
+            if other_node not in self.__already_visited_node__:
+                related_entities.update(self.get_related_entities(other_node))
+
+        return self.__all_connections__.get(entity_id).union(related_entities)
+
+    def get_queryset(self):
+        # Get related entities from source
+        source_node = Entity.objects.none()
+
+        try:
+            entity_uid = self.kwargs.get("uid")
+            source_node = Entity.objects.get(uid=entity_uid)
+        except ObjectDoesNotExist:
+            raise Http404
```

Comment:
I think this could be simplified into: 

```suggestion
        source_node = get_object_or_404(Entity, uid=self.kwargs.get("uid"))
```

See: https://docs.djangoproject.com/en/4.0/topics/http/shortcuts/#get-object-or-404

⚠️ needs the import too if you accept the suggestion

Reply:
Nice, I should have think about this the first time